OrbitQt/topdownwidget.cpp

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "topdownwidget.h"
+
+#include <QMenu>
+
+#include "TopDownViewItemModel.h"
+
+void TopDownWidget::SetTopDownView(std::unique_ptr<TopDownView> top_down_view) {
+  auto* model =
+      new TopDownViewItemModel{std::move(top_down_view), ui_->topDownTreeView};
+  auto* proxy_model = new QSortFilterProxyModel{ui_->topDownTreeView};
+  proxy_model->setSourceModel(model);
+  proxy_model->setSortRole(Qt::EditRole);
+  ui_->topDownTreeView->setModel(proxy_model);
+  ui_->topDownTreeView->sortByColumn(TopDownViewItemModel::kInclusive,
+                                     Qt::DescendingOrder);
+  for (int column = 0; column < TopDownViewItemModel::kColumnCount; ++column) {
+    ui_->topDownTreeView->resizeColumnToContents(column);
```

Comment:
This doesn't seem to work for the name column on my side, unless there is a maximum width specified somewhere?

Reply:
I think that this works, but not like we'd like because the resizing is done based on the visible elements, which when this code is called are only the thread names. I'm not sure what a good way of doing this is.

Reply:
You can use QHeaderView to achieve what you want (probably). Have a look in one of my branches: https://github.com/beckerhe/orbitprofiler/blob/9e362d1d4611c88ada629c57002cb71ca74b0227/OrbitQt/orbitmainwindow.cpp#L270

Reply:
And a doc link: https://doc.qt.io/qt-5/qheaderview.html#setSectionResizeMode-1

Reply:
I had actually tried that but if I use `Stretch` the behavior is weird. The section is not manually resizable anymore. You can resize it indirectly by resizing the other ones but as it's the first section (or rather, not the last) the overall behavior is annoying.